### PR TITLE
[chore] excluded filters section

### PR DESCRIPTION
**Pull request by Kharonus**
- added section to exclude filters from schema dependeny collection

### CODE REVIEW

File: lib/api/v3/queries/schemas/query_schema_representer.rb [hunk 1]
```diff
@@ -283,6 +282,15 @@ def filter_instance_schemas_href
               api_v3_paths.query_filter_instance_schemas
             end
           end
+
+          private
+
+          def excluded_filters
+            # TODO: The RelatableFilter is not supported by the schema dependencies yet
+            [
+              ::Queries::WorkPackages::Filter::RelatableFilter
```
Comment (Kharonus) [inline lib/api/v3/queries/schemas/query_schema_representer.rb]: alternatively we could define a method lookup in the filter itself, that is called from here?

File: lib/api/v3/queries/schemas/query_filter_instance_schema_representer.rb [hunk 2]
```diff
@@ -126,7 +124,7 @@ def dependencies
                                                                 operator,
                                                                 form_embedded: form_embedded)
 
-              hash[path] = value
+              hash[path] = value unless value.nil?
```
Comment (Kharonus) [inline lib/api/v3/queries/schemas/query_filter_instance_schema_representer.rb:127]: This is the place, where the exclusion takes place, when the factory returns `nil`. It is not written to the hash, hence, it does not appear in the response.

File: lib/api/v3/queries/schemas/filter_dependency_representer_factory.rb [hunk 3]
```diff
@@ -52,21 +52,25 @@ def create(filter, operator, form_embedded: false)
...
+            UpdatedAtFilter: 'DateTimeFilter',
+            AuthorFilter: 'UserFilter',
+            ResponsibleFilter: 'AllPrincipalsFilter',
+            AssignedToFilter: 'AllPrincipalsFilter',
+            WatcherFilter: 'UserFilter'
           }
 
+          @excluded_filters = [
+            ::Queries::WorkPackages::Filter::RelatableFilter
+          ]
```
Comment (ulferts) [inline lib/api/v3/queries/schemas/filter_dependency_representer_factory.rb:65]: If you want to exclude filters from within a module e.g. storage we will need to add a method for doing so. Don't know if you want to do this within the scope of this PR, @Kharonus.

File: lib/api/v3/queries/schemas/filter_dependency_representer_factory.rb [hunk 4]
```diff
@@ -52,21 +52,25 @@ def create(filter, operator, form_embedded: false)
...
             name = filter_specific_representer_class(filter) ||
-                   cf_representer_class(filter) ||
-                   type_specific_representer_class(filter) ||
-                   custom_representer_class(filter)
+              cf_representer_class(filter) ||
+              type_specific_representer_class(filter) ||
+              custom_representer_class(filter)
 
-            name.constantize
+            name.nil? ? nil : name.constantize
```
Comment (ulferts) [inline lib/api/v3/queries/schemas/filter_dependency_representer_factory.rb:73]: You could use the safe navigation operator here: `name&.constantize`

**Review by ulferts — COMMENTED** [review]
Is good to merge IMO. Just raised a question. Feel free to merge @Kharonus .

File: lib/api/v3/queries/schemas/filter_dependency_representer_factory.rb (hunk 3 above)
Comment (Kharonus) [reply lib/api/v3/queries/schemas/filter_dependency_representer_factory.rb:65]: How is it different, when the filter is defined in a module?

File: lib/api/v3/queries/schemas/filter_dependency_representer_factory.rb (hunk 4 above)
Comment (Kharonus) [reply lib/api/v3/queries/schemas/filter_dependency_representer_factory.rb:73]: true, will change it

File: lib/api/v3/queries/schemas/filter_dependency_representer_factory.rb (hunk 3 above)
Comment (ulferts) [reply lib/api/v3/queries/schemas/filter_dependency_representer_factory.rb:65]: In order to keep the separation I'd expect some sort of registration mechanism in the module engine, e.g. 

```
config.to_prepare do
 API::...::FilterDependencyRepresenterFactory.add_excluded_filters ::MyFilter
end
```

Comment (Kharonus) [reply lib/api/v3/queries/schemas/filter_dependency_representer_factory.rb:65]: Ah, so you wouldn't add it directly into the factory but have a register hook. Agreed.

**Comment by cbliard** [comment]
I merged it to make the feature/file-links branch green again. We can address ulferts remarks later.

**Pull request merged**